skll/metrics.py

```diff
@@ -164,14 +204,30 @@ def kendall_tau(y_true, y_pred):
     ret_score : float
         Kendall's tau if well-defined, else 0.0
     """
-    ret_score = kendalltau(y_true, y_pred)[0]
+    # convert to numpy array in case we are passed a list
+    y_pred = np.array(y_pred)
+
+    # multi-dimensional -> probability array -> get label
+    if y_pred.ndim > 1:
+        labels = np.argmax(y_pred, axis=1)
+        ret_score = kendalltau(y_true, labels)[0]
+    # 1-dimensional -> probabilities/labels -> use as is
+    else:
+        ret_score = kendalltau(y_true, y_pred)[0]
     return ret_score if not np.isnan(ret_score) else 0.0
 
 
 def spearman(y_true, y_pred):
```

Comment:
Hmm, I didn't really think about that. I am not sure how easy or readable it would be since we these are essentially being passed to `make_scorer()` but I will try and take a look.

Reply:
It wasn't that hard actually and it does cut down on duplication! Thanks for the suggestion!